event_model/__init__.py

```diff
@@ -1625,7 +1625,7 @@ def compose_descriptor(*, start, streams, event_counter,
         partial(compose_event_page, descriptor=doc, event_counter=event_counter))
 
 
-def compose_run(*, uid=None, time=None, metadata=None, validate=True):
+def compose_run(*, uid=None, time=None, metadata=None, validate=True, projections=None):
```

Comment:
All other non-required keys enter in through `metadata`. I think we should either add all of them to the signature or leave `projections` out of the signature.

Reply:
I'll back out that change, let you add a projection to metadata.